spot_driver/spot_driver/spot_ros2.py

```diff
@@ -844,6 +844,31 @@ def __init__(self, parameter_list: Optional[typing.List[Parameter]] = None, **kw
                 if self.auto_stand.value:
                     self.spot_wrapper.stand()
 
+        # Explicitly get a lease when get_lease_on_action is False
+        # because spot_wrapper does not get a lease at initialization.
+        if self.spot_wrapper is not None and not self.get_lease_on_action.value:
+            self.spot_wrapper.getLease()
+        self.create_service(
+            srv_type=Trigger,
+            srv_name="gen_new_spot_ros_lease",
+            callback=self.new_lease_callback,
+            callback_group=self.group,
+        )
+
+    def new_lease_callback(self, request: Trigger.Request, response: Trigger.Response) -> Trigger.Response:
+        self.get_logger().info("Incoming request to generate a new lease.")
+        if self.spot_wrapper is None:
+            response.success = True
+            response.message = "SpotROS is running in mock mode."
+            return response
+
+        old_lease = self.spot_wrapper._lease
+        self.spot_wrapper.getLease()
+        lease = self.spot_wrapper._lease
+        response.success = str(lease.lease_proto) != str(old_lease.lease_proto)
```

Comment:
Does the BD API itself provide a notion of success/failure here instead of trying to define a new notion of success? How would `self.spot_wrapper._lease` not change after a call to `self.spot_wrapper.getLease()`? I think only if the BD python API throws one of the exceptions corresponding to the error responses:

https://dev.bostondynamics.com/protos/bosdyn/api/proto_reference.html#bosdyn-api-TakeLeaseResponse-Status

https://dev.bostondynamics.com/protos/bosdyn/api/proto_reference.html#bosdyn-api-AcquireLeaseResponse-Status

What happens when one of the exceptions is thrown?

Reply:
https://dev.bostondynamics.com/_modules/bosdyn/client/lease#LeaseResponseError

`LeaseClient.take()` can raise `InvalidResourceError`, or `NotAuthoritativeServiceError`. We don't encounter the former, because if we have no argument to `take()`, as in the call inside `SpotWrapper.getLease()`, it defaults to 'body' which I think is always a valid resource. 

And I have no idea when the latter comes up -- I don't understand it. 

So yeah, I can put this in a try/except if that is preferred. 

I think I was thinking that someone might change `SpotWrapper.getLease` at some point in a way that returns the current lease and not necessarily a new lease, and only checking for these exceptions would not catch that. I at least want to assert that the lease returned by `SpotWrapper.getLease()` is different than the lease before the call to that method.